nixos/modules/services/networking/fedimintd.nix

```diff
@@ -0,0 +1,357 @@
+        };
+        port = mkOption {
+          type = types.port;
+          default = 8174;
+          description = "Port to bind on for API connections relied by the reverse proxy/tls terminator.";
+        };
+        bind = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = "Address to bind on for API connections relied by the reverse proxy/tls terminator. Usually starting with `fedimint://`";
+        };
+        fqdn = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "api.myfedimint.com";
+          description = "Public domain of the API address of the reverse proxy/tls terminator.";
+        };
+        address = mkOption {
+          type = types.nullOr types.str;
+          default = if config.api.fqdn != null then "wss://${config.api.fqdn}" else null;
+          description = ''
+            Public URL of the API address of the reverse proxy/tls terminator. Usually starting with `wss://`.
+
+            Typically you want to override `fqdn` instead.
+          '';
+        };
+      };
+      bitcoin = {
+        network = mkOption {
+          type = types.str;
+          default = "signet";
+          example = "bitcoin";
+          description = "Bitcoin network to participate in.";
+        };
+        rpc = {
+          address = mkOption {
+            type = types.str;
+            default = "http://127.0.0.1:38332";
+            example = "signet";
+            description = "Bitcoin node (bitcoind/electrum/esplora) address to connect to";
+          };
+
+          kind = mkOption {
+            type = types.str;
+            default = "bitcoind";
+            example = "electrum";
+            description = "Kind of a bitcoin node.";
+          };
+
+          secretFile = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = ''
+              If set the URL specified in `bitcoin.rpc.address` will get the content of this file added
+              as an URL password, so `http://user@example.com` will turn into `http://user:SOMESECRET@example.com`.
+
+              Example:
+
+              `/etc/nix-bitcoin-secrets/bitcoin-rpcpassword-public` (for nix-bitcoin default)
+            '';
+          };
+        };
+      };
+
+      consensus.finalityDelay = mkOption {
+        type = types.number;
+        default = 10;
+        description = "Consensus peg-in finality delay.";
+      };
+
+      dataDir = mkOption {
+        type = types.str;
+        default = "/var/lib/fedimintd-${name}/";
+        readOnly = true;
+        description = ''
+          Path to the data dir fedimintd will use to store its data.
+          Note that due to using the DynamicUser feature of systemd, this value should not be changed
+          and is set to be read only.
+        '';
+      };
+
+      nginx = {
+        enable = mkEnableOption "fedimint";
```

Comment:
We should probably change the description here to `Wether to configure nginx for fedimint` or similar

Reply:
IIUC no other way than to convert to `mkOption` for that, so did that.